src/utils/client.js

```diff
@@ -273,14 +283,22 @@ export default class Client {
     });
   }
 
-  submitCustom(customKeyValues) {
+  async submitCustom(customKeyValues) {
     const { userKey } = initInfo();
     const url = `${API_URL}panel/custom/submit/v2/`;
 
     const jsonKeyVals = {};
     const formData = new FormData();
-    Object.entries(customKeyValues).forEach(([key, val]) => {
+    const promiseResults = await Promise.all(
+      Object.entries(customKeyValues).map(([key, val]) =>
+        Promise.all([key, Promise.resolve(val)])
+      )
+    );
+    promiseResults.forEach(([key, val]) => {
       if (val instanceof Blob) {
+        // If you use val from customKeyValues, instead of value from
```

Comment:
first comma here is unnecessary i think and threw me off a little. can you explain why you can't use val from customKeyValues?

Reply:
I removed the first comma. The second sentence explains why it doesn't work - the files aren't resolved before submission. I confirmed that the original code for this function does *not* actually submit files correctly to the BE

Reply:
ah cool makes sense